extensions/common/context.cc

```diff
@@ -227,6 +227,56 @@ void populateHTTPRequestInfo(bool outbound, RequestInfo* request_info) {
   request_info->destination_port = destination_port;
 }
 
+void populateTCPRequestInfo(bool outbound, RequestInfo* request_info) {
+  // TODO: switch to stream_info.requestComplete() to avoid extra compute.
+  request_info->end_timestamp = getCurrentTimeNanoseconds();
+
+  // Fill in request info.
+  request_info->response_code = 0;
+  request_info->request_protocol = "tcp";
+
+  // Try to get fqdn of destination service from cluster name. If not found, use
+  // host header instead.
+  std::string cluster_name = "";
+  getStringValue({"cluster_name"}, &cluster_name);
+  extractFqdn(cluster_name, &request_info->destination_service_host);
+  if (!request_info->destination_service_host.empty()) {
+    // cluster name follows Istio convention, so extract out service name.
+    extractServiceName(request_info->destination_service_host,
+                       &request_info->destination_service_name);
+  }
+
+  // Get rbac labels from dynamic metadata.
+  getStringValue({"metadata", kRbacFilterName, kRbacPermissivePolicyIDField},
+                 &request_info->rbac_permissive_policy_id);
+  getStringValue(
+      {"metadata", kRbacFilterName, kRbacPermissiveEngineResultField},
+      &request_info->rbac_permissive_engine_result);
+
+  int64_t destination_port = 0;
+
+  if (outbound) {
+    getValue({"upstream", "port"}, &destination_port);
+    getStringValue({"upstream", "uri_san_peer_certificate"},
+                   &request_info->destination_principal);
+    getStringValue({"upstream", "uri_san_local_certificate"},
+                   &request_info->source_principal);
+  } else {
+    // getValue({"destination", "port"}, &destination_port);
```

Comment:
Why is this commented out?

Reply:
This was giving segmentation fault for TCP, hence commented out for now.

Reply:
Actually, just looked downstream address is filled in tcp proxy => https://cs.corp.google.com/piper///depot/google3/third_party/envoy/src/source/common/tcp_proxy/tcp_proxy.cc?sq=package:piper+file://depot/google3+-file:google3/(experimental%7Cobsolete)&type=cs&rcl=279545696&g=0&l=208

Stats filter currently, runs before tcp proxy... should it actually run after?

Reply:
Actually, just looked downstream address is filled in tcp proxy and stats filter currently, runs before tcp proxy... should it actually run after?

Reply:
Are filter after TCP proxy even executed? TCP proxy writes directly to the connection object, so I'm not sure if you'd be getting correct callbacks if you install it after TCP proxy.

Reply:
Yeah, just tried.. filters are not allowed after TCP Proxy... debugging inside wasm to see why I am getting segfault here..

Reply:
Looks like local destination address is null.(downstreamLocalAddress).. StreamInfo says this can never be null though...

Reply:
Would it be possible to share the code between this and the one above? There's a lot of duplication.

Reply:
Done..